The new test passes with the receiver enabled. The failure now seems unrelated - 

```
--- FAIL: TestExporter_PushEvent (0.87s)
    --- FAIL: TestExporter_PushEvent/no_retry (0.00s)
        --- FAIL: TestExporter_PushEvent/no_retry/fail_http_request (0.00s)
            --- FAIL: TestExporter_PushEvent/no_retry/fail_http_request/max_requests_limited (0.20s)
                logger.go:130: 2021-03-30T02:22:26.647Z	DEBUG	Request roundtrip completed.	{"path": "/_bulk", "method": "POST", "duration": "2.509125ms", "status": "500 Internal Server Error"}
                logger.go:130: 2021-03-30T02:22:26.650Z	DEBUG	Request roundtrip completed.	{"path": "/_bulk", "method": "POST", "duration": "715.242µs", "status": "500 Internal Server Error"}
                logger.go:130: 2021-03-30T02:22:26.651Z	ERROR	Bulk indexer error: flush: %!s(<nil>)
                logger.go:130: 2021-03-30T02:22:26.652Z	ERROR	Bulk indexer error: flush: [500 Internal Server Error] oops
                exporter_test.go:212: 
                    	Error Trace:	exporter_test.go:212
                    	Error:      	Not equal: 
                    	            	expected: 1
                    	            	actual  : 2
                    	Test:       	TestExporter_PushEvent/no_retry/fail_http_request/max_requests_limited
            --- FAIL: TestExporter_PushEvent/no_retry/fail_http_request/retry.enabled_is_false (0.20s)
                logger.go:130: 2021-03-30T02:22:26.647Z	DEBUG	Request roundtrip completed.	{"path": "/_bulk", "method": "POST", "duration": "1.929137ms", "status": "500 Internal Server Error"}
                logger.go:130: 2021-03-30T02:22:26.650Z	DEBUG	Request roundtrip completed.	{"path": "/_bulk", "method": "POST", "duration": "725.017µs", "status": "500 Internal Server Error"}
                logger.go:130: 2021-03-30T02:22:26.651Z	ERROR	Bulk indexer error: flush: %!s(<nil>)
                logger.go:130: 2021-03-30T02:22:26.652Z	ERROR	Bulk indexer error: flush: [500 Internal Server Error] oops
                exporter_test.go:212: 
                    	Error Trace:	exporter_test.go:212
                    	Error:      	Not equal: 
                    	            	expected: 1
                    	            	actual  : 2
                    	Test:       	TestExporter_PushEvent/no_retry/fail_http_request/retry.enabled_is_false
FAIL
coverage: 85.7% of statements
FAIL	github.com/open-telemetry/opentelemetry-collector-contrib/exporter/elasticexporter	1.091s
FAIL
make[2]: *** [../../Makefile.Common:48: do-unit-tests-with-cover] Error 1
make[2]: Leaving directory '/home/circleci/project/exporter/elasticsearchexporter'
make[1]: *** [Makefile:81: for-all] Error 2
make[1]: Leaving directory '/home/circleci/project'
make: *** [Makefile:47: unit-tests-with-cover] Error 2
```

--- a/testbed/datareceivers/splunk.go
+++ b/testbed/datareceivers/splunk.go
@@ -1,0 +1,79 @@
+// Copyright 2020 OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package datareceivers
+
+import (
+	"context"
+	"fmt"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/config/confighttp"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/testbed/testbed"
+	"go.uber.org/zap"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/splunkhecreceiver"
+)
+
+// SplunkHECDataReceiver implements Splunk HEC format receiver.
+type SplunkHECDataReceiver struct {
+	testbed.DataReceiverBase
+	receiver component.LogsReceiver
+}
+
+// Ensure SplunkHECDataReceiver implements LogDataSender.
+var _ testbed.DataReceiver = (*SplunkHECDataReceiver)(nil)
+
+// NewSplunkHECDataReceiver creates a new SplunkHECDataReceiver that will listen on the
+// specified port after Start is called.
+func NewSplunkHECDataReceiver(port int) *SplunkHECDataReceiver {
+	return &SplunkHECDataReceiver{DataReceiverBase: testbed.DataReceiverBase{Port: port}}
+}
+
+// Start the receiver.
+func (sr *SplunkHECDataReceiver) Start(_ consumer.Traces, _ consumer.Metrics, lc consumer.Logs) error {
+	config := splunkhecreceiver.Config{
+		HTTPServerSettings: confighttp.HTTPServerSettings{
+			Endpoint: fmt.Sprintf("localhost:%d", sr.Port),
+		},
+	}
+	var err error
+	f := splunkhecreceiver.NewFactory()
+	sr.receiver, err = f.CreateLogsReceiver(context.Background(), component.ReceiverCreateParams{Logger: zap.L()}, &config, lc)
+	if err != nil {
+		return err
+	}
+
+	return sr.receiver.Start(context.Background(), sr)
+}
+
+// Stop the receiver.
+func (sr *SplunkHECDataReceiver) Stop() error {
+	return sr.receiver.Shutdown(context.Background())
+}
+
+// GenConfigYAMLStr returns exporter config for the agent.
+func (sr *SplunkHECDataReceiver) GenConfigYAMLStr() string {
+	// Note that this generates an exporter config for agent.
+	return fmt.Sprintf(`
+    splunk_hec:
+      endpoint: "http://localhost:%d"
+      token: "token"`, sr.Port)
+}
+
+// ProtocolName returns protocol name as it is specified in Collector config.
+func (sr *SplunkHECDataReceiver) ProtocolName() string {
+	return "splunk_hec"
+}

--- a/testbed/datasenders/fluent.go
+++ b/testbed/datasenders/fluent.go
@@ -1,0 +1,167 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package datasenders
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"strconv"
+	"testing"
+	"time"
+
+	"github.com/fluent/fluent-logger-golang/fluent"
+	"github.com/stretchr/testify/require"
+	"go.opentelemetry.io/collector/consumer/pdata"
+	"go.opentelemetry.io/collector/testbed/testbed"
+)
+
+const (
+	fluentDatafileVar = "FLUENT_DATA_SENDER_DATA_FILE"
+	fluentPortVar     = "FLUENT_DATA_SENDER_RECEIVER_PORT"
+)
+
+// FluentLogsForwarder forwards logs to fluent forwader
+type FluentLogsForwarder struct {
+	testbed.DataSenderBase
+	fluentLogger *fluent.Fluent
+	dataFile     *os.File
+}
+
+// Ensure FluentLogsForwarder implements LogDataSender.
+var _ testbed.LogDataSender = (*FluentLogsForwarder)(nil)
+
+func NewFluentLogsForwarder(t *testing.T, port int) *FluentLogsForwarder {
+	var err error
+	portOverride := os.Getenv(fluentPortVar)
+	if portOverride != "" {
+		port, err = strconv.Atoi(portOverride)
+		require.NoError(t, err)
+	}
+
+	f := &FluentLogsForwarder{DataSenderBase: testbed.DataSenderBase{Port: port}}
+
+	// When FLUENT_DATA_SENDER_DATA_FILE is set, the data sender, writes to a
+	// file. This enables users to optionally run the e2e test against a real
+	// fluentd/fluentbit agent rather than using the fluent writer the data sender
+	// uses by default. In case, one is looking to point a real fluentd/fluentbit agent
+	// to the e2e test, they can do so by configuring the fluent agent to read from the
+	// file FLUENT_DATA_SENDER_DATA_FILE and forward data to FLUENT_DATA_SENDER_RECEIVER_PORT
+	// on 127.0.0.1.
+	if dataFileName := os.Getenv(fluentDatafileVar); dataFileName != "" {
+		f.dataFile, err = os.OpenFile(dataFileName, os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0600)
+		require.NoError(t, err)
+	} else {
+		logger, err := fluent.New(fluent.Config{FluentPort: port, Async: true})
+		require.NoError(t, err)
+		f.fluentLogger = logger
+	}
+	return f
+}
+
+func (f *FluentLogsForwarder) Start() error {
+	return nil
+}
+
+func (f *FluentLogsForwarder) Stop() error {
+	return f.fluentLogger.Close()
+}
+
+func (f *FluentLogsForwarder) ConsumeLogs(_ context.Context, logs pdata.Logs) error {
+	for i := 0; i < logs.ResourceLogs().Len(); i++ {
+		for j := 0; j < logs.ResourceLogs().At(i).InstrumentationLibraryLogs().Len(); j++ {
+			ills := logs.ResourceLogs().At(i).InstrumentationLibraryLogs().At(j)
+			for k := 0; k < ills.Logs().Len(); k++ {
+				if f.dataFile == nil {
+					if err := f.fluentLogger.Post("", f.convertLogToMap(ills.Logs().At(k))); err != nil {
+						return err
+					}
+				} else {
+					if _, err := f.dataFile.Write(append(f.convertLogToJSON(ills.Logs().At(k)), '\n')); err != nil {
+						return err
+					}
+				}
+			}
+		}
+	}
+	return nil
+}
+
+func (f *FluentLogsForwarder) convertLogToMap(lr pdata.LogRecord) map[string]string {
+	out := map[string]string{}
+
+	if lr.Body().Type() == pdata.AttributeValueSTRING {
+		out["log"] = lr.Body().StringVal()
+	}
+
+	lr.Attributes().ForEach(func(k string, v pdata.AttributeValue) {
+		switch v.Type() {
+		case pdata.AttributeValueSTRING:
+			out[k] = v.StringVal()
+		case pdata.AttributeValueINT:
+			out[k] = strconv.FormatInt(v.IntVal(), 10)
+		case pdata.AttributeValueDOUBLE:
+			out[k] = strconv.FormatFloat(v.DoubleVal(), 'f', -1, 64)
+		case pdata.AttributeValueBOOL:
+			out[k] = strconv.FormatBool(v.BoolVal())
+		default:
+			panic("missing case")
+		}
+	})
+
+	return out
+}
+
+func (f *FluentLogsForwarder) convertLogToJSON(lr pdata.LogRecord) []byte {
+	rec := map[string]string{
+		"time": time.Unix(0, int64(lr.Timestamp())).Format("02/01/2006:15:04:05Z"),
+	}
+	rec["log"] = lr.Body().StringVal()
+
+	lr.Attributes().ForEach(func(k string, v pdata.AttributeValue) {
+		switch v.Type() {
+		case pdata.AttributeValueSTRING:
+			rec[k] = v.StringVal()
+		case pdata.AttributeValueINT:
+			rec[k] = strconv.FormatInt(v.IntVal(), 10)
+		case pdata.AttributeValueDOUBLE:
+			rec[k] = strconv.FormatFloat(v.DoubleVal(), 'f', -1, 64)
+		case pdata.AttributeValueBOOL:
+			rec[k] = strconv.FormatBool(v.BoolVal())
+		default:
+			panic("missing case")
+		}
+	})
+	b, err := json.Marshal(rec)
+	if err != nil {
+		panic("failed to write log: " + err.Error())
+	}
+	return b
+}
+
+func (f *FluentLogsForwarder) Flush() {
+	_ = f.dataFile.Sync()
+}
+
+func (f *FluentLogsForwarder) GenConfigYAMLStr() string {
+	return fmt.Sprintf(`
+  fluentforward:
+    endpoint: localhost:%d`, f.Port)
+}
+
+func (f *FluentLogsForwarder) ProtocolName() string {
+	return "fluentforward"
+}

--- a/testbed/go.mod
+++ b/testbed/go.mod
@@ -3,6 +3,7 @@ module github.com/open-telemetry/opentelemetry-collector-contrib/testbed
 go 1.14
 
 require (
+	github.com/fluent/fluent-logger-golang v1.5.0
 	github.com/open-telemetry/opentelemetry-collector-contrib/exporter/carbonexporter v0.0.0-00010101000000-000000000000
 	github.com/open-telemetry/opentelemetry-collector-contrib/exporter/sapmexporter v0.0.0-00010101000000-000000000000
 	github.com/open-telemetry/opentelemetry-collector-contrib/exporter/signalfxexporter v0.0.0-00010101000000-000000000000
@@ -10,7 +11,10 @@ require (
 	github.com/open-telemetry/opentelemetry-collector-contrib/receiver/carbonreceiver v0.0.0-00010101000000-000000000000
 	github.com/open-telemetry/opentelemetry-collector-contrib/receiver/sapmreceiver v0.0.0-00010101000000-000000000000
 	github.com/open-telemetry/opentelemetry-collector-contrib/receiver/signalfxreceiver v0.0.0-00010101000000-000000000000
+	github.com/open-telemetry/opentelemetry-collector-contrib/receiver/splunkhecreceiver v0.0.0-00010101000000-000000000000
 	github.com/open-telemetry/opentelemetry-collector-contrib/testbed/mockdatareceivers/mockawsxrayreceiver v0.0.0-00010101000000-000000000000
+	github.com/stretchr/testify v1.7.0
+	github.com/tinylib/msgp v1.1.5 // indirect
 	go.opentelemetry.io/collector v0.23.1-0.20210331155714-fe86ab5cba6d
 	go.uber.org/zap v1.16.0
 )
@@ -20,6 +24,8 @@ replace github.com/open-telemetry/opentelemetry-collector-contrib/exporter/carbo
 replace github.com/open-telemetry/opentelemetry-collector-contrib/exporter/sapmexporter => ../exporter/sapmexporter
 
 replace github.com/open-telemetry/opentelemetry-collector-contrib/exporter/signalfxexporter => ../exporter/signalfxexporter
+
+replace github.com/open-telemetry/opentelemetry-collector-contrib/exporter/splunkhecexporter => ../exporter/splunkhecexporter
 
 replace github.com/open-telemetry/opentelemetry-collector-contrib/internal/common => ../internal/common
 
@@ -34,5 +40,7 @@ replace github.com/open-telemetry/opentelemetry-collector-contrib/receiver/carbo
 replace github.com/open-telemetry/opentelemetry-collector-contrib/receiver/sapmreceiver => ../receiver/sapmreceiver
 
 replace github.com/open-telemetry/opentelemetry-collector-contrib/receiver/signalfxreceiver => ../receiver/signalfxreceiver
+
+replace github.com/open-telemetry/opentelemetry-collector-contrib/receiver/splunkhecreceiver => ../receiver/splunkhecreceiver
 
 replace github.com/open-telemetry/opentelemetry-collector-contrib/testbed/mockdatareceivers/mockawsxrayreceiver => ../testbed/mockdatareceivers/mockawsxrayreceiver

--- a/testbed/tests/log_test.go
+++ b/testbed/tests/log_test.go
@@ -23,6 +23,7 @@ import (
 	"go.opentelemetry.io/collector/testbed/testbed"
 	scenarios "go.opentelemetry.io/collector/testbed/tests"
 
+	"github.com/open-telemetry/opentelemetry-collector-contrib/testbed/datareceivers"
 	"github.com/open-telemetry/opentelemetry-collector-contrib/testbed/datasenders"
 )
 
@@ -125,6 +126,15 @@ func TestLog10kDPS(t *testing.T) {
 				ExpectedMaxRAM: 155,
 			},
 			extensions: flw.Extensions(),
+		},
+		{
+			name:     "FluentForward-SplunkHEC",
+			sender:   datasenders.NewFluentLogsForwarder(t, testbed.GetAvailablePort(t)),
+			receiver: datareceivers.NewSplunkHECDataReceiver(testbed.GetAvailablePort(t)),
+			resourceSpec: testbed.ResourceSpec{
+				ExpectedMaxCPU: 60,
+				ExpectedMaxRAM: 150,
+			},
 		},
 	}
 

